pkg/detector/ospkg/oracle/oracle.go

```diff
@@ -44,6 +44,18 @@ func NewScanner() *Scanner {
 	}
 }
 
+func extractKspliceNumber(v string) string {
+	subs := strings.Split(strings.ToLower(v), "ksplice")
+	if len(subs) != 2 {
+		return ""
+	}
+	ns := strings.Split(subs[1], ".")
+	if len(ns) == 0 {
+		return ""
+	}
+	return ns[0]
+}
+
```

Comment:
If we can assume ksplice is separated by dot, how about this?

```suggestion
func extractKspliceNumber(v string) string {
	subs := strings.Split(strings.ToLower(v), ".")
	for _, s := range subs {
		if strings.HasPrefix(v, "ksplice") {
			return s
		}
	}
	return ""
}

```